File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const newSet = new Set(nums);
        let count = {};
        for(let i = 0; i < nums.length; i++) {
            for(let j = 1; j <= newSet.length; j++) {
                if(!count[j]) {
                    count[j] = [nums[i]];
                } else {
                    count[j].push(nums[i])
                }
            }
        }
        return count[k];
        // 1 2 2 3 3 3
        // 1 : [1, 2, 3]
        // 2 : [2, 3]
        // 3 : [3]
        // ...
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [0] * (n+2)
        for i in range(2, n+2):
            dp[i] = dp[i-1] + dp[i-2]
        return dp[-1]